### PR TITLE
[processor/k8sattributes] Move config validation into Validate

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** 
Moves existing config validation into the Validate function

**Testing:** 
Added more unit tests

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
Found a couple more scenarios to move into `Validate`, stay tuned.

**Comment by TylerHelmuth** [comment]
@dmitryax I believe I found all validation instances in `options.go` and moved them to config's validate.  Ready for a re-review.

File: processor/k8sattributesprocessor/options.go [hunk 1]
```diff
@@ -235,11 +227,7 @@ func extractFieldRules(fieldType string, fields ...FieldExtractConfig) ([]kube.F
 			var err error
 			r, err = regexp.Compile(a.Regex)
 			if err != nil {
-				return rules, err
-			}
-			names := r.SubexpNames()
-			if len(names) != 2 || names[1] != "value" {
-				return rules, fmt.Errorf("regex must contain exactly one named submatch (value)")
+				return rules, fmt.Errorf("this is an error in the k8sattributesprocessor, %w", err)
```
Comment (dmitryax) [inline processor/k8sattributesprocessor/options.go:230]: I don't think this line adds any info. Collector should add component name filed to every error
```suggestion
				return rules, err
```

File: processor/k8sattributesprocessor/options.go [hunk 2]
```diff
@@ -249,7 +237,7 @@ func extractFieldRules(fieldType string, fields ...FieldExtractConfig) ([]kube.F
 			var err error
 			keyRegex, err = regexp.Compile("^(?:" + a.KeyRegex + ")$")
 			if err != nil {
-				return rules, err
+				return rules, fmt.Errorf("this is an error in the k8sattributesprocessor, %w", err)
```
Comment (dmitryax) [inline processor/k8sattributesprocessor/options.go:240]: Same here

File: processor/k8sattributesprocessor/options.go (hunk 1 above)
Comment (TylerHelmuth) [reply processor/k8sattributesprocessor/options.go:230]: Technically we could only get here if something we wrong with the validate function. Another option is to ignore the error and then remove it from the option interface.

Comment (dmitryax) [reply processor/k8sattributesprocessor/options.go:230]: I think just returning error as is a better option

File: processor/k8sattributesprocessor/options.go (hunk 2 above)
Comment (TylerHelmuth) [inline processor/k8sattributesprocessor/options.go:240]:
```suggestion
				return rules, err
```

**Review by dmitryax — APPROVED** [review]

**Pull request merged**